Applications/CMC/cmc.cpp

```diff
@@ -111,20 +116,36 @@ int main(int argc,char **argv)
 
     // CONSTRUCT
     cout<<"Constructing investigation from setup file "<<setupFileName<<".\n"<<endl;
-    CMCTool cmcgait(setupFileName);
-
-    // PRINT MODEL INFORMATION
-    Model& model = cmcgait.getModel();
-    cout<<"-----------------------------------------------------------------------\n";
-    cout<<"Loaded library\n";
-    cout<<"-----------------------------------------------------------------------\n";
-    model.printBasicInfo(cout);
-    cout<<"-----------------------------------------------------------------------\n\n";
-
-
-    // RUN
-    cmcgait.run();
-
+    if(specifiedMaxNumThreads != -1)
+    {
+        CMCTool cmcgait(setupFileName,specifiedMaxNumThreads);
+        
```

Comment:
This is the only statement that differs in the `if` and `else` blocks so it is the only statement that belongs there. Also if there was a default `specifiedMaxNumThreads` = 1 then you wouldn't even need the `if`.